sql-cli/sql_cli/dag_runner.py

```diff
@@ -48,6 +48,39 @@ def _local_connections(self) -> dict[str, Connection]:
         return conns
 
 
+def _update_secrets(
+    operation: str,
+    conn_file_path: str | None = None,
+    variable_file_path: str | None = None,
+    connections: dict[str, Connection] | None = None,
+):
+    """Update secrets in secrets backend by fetching connections based on given arguments"""
+    if conn_file_path or variable_file_path or connections:
+        # To get around the fact that we reload certain modules, we need to import here
+        from airflow.configuration import secrets_backend_list
+
+        if operation == "add":
+            local_secrets = AstroFilesystemBackend(
+                variables_file_path=variable_file_path,
+                connections_file_path=conn_file_path,
+                connections=connections,
+            )
+            secrets_backend_list.insert(0, local_secrets)
+        elif operation == "remove":
+            # Remove the local variables we have added to the secrets_backend_list
+            secrets_backend_list.pop(0)  # noqa
+
+
+def _get_tailored_dag(dag: DAG, task_id: str | None, include_upstream: bool) -> DAG:
+    """Gets subset of the dag based on whether a task ID is given and asked to include its upstream tasks in the dag."""
+    if task_id and include_upstream:
+        dag = dag.partial_subset(
+            task_ids_or_regex=task_id,
+            include_upstream=True,
+        )
+    return dag
```

Comment:
Can you explain what new functionality is happening here? I'd like to know why you broke these parts off.

Reply:
We broke these into multiple methods with the addition of a few new conditionals like https://github.com/astronomer/astro-sdk/pull/1766/files#diff-468cfaa4c543746918f04d1238ff10da015a5ecdae4b58178f66150b5b6c65cdR148  because ruff was complaining that the earlier function was too complex.